float64[] goal
float64 drag_goal
float64 penetrate_goal
float64 distance_goal
---
bool finish